presto-spi/src/main/java/com/facebook/presto/spi/block/SingleMapBlockEncoding.java

```diff
@@ -61,23 +51,30 @@ public String getName()
     }
 
     @Override
-    public void writeBlock(SliceOutput sliceOutput, Block block)
+    public void writeBlock(BlockEncodingSerde blockEncodingSerde, SliceOutput sliceOutput, Block block)
     {
         SingleMapBlock singleMapBlock = (SingleMapBlock) block;
+        TypeSerde.writeType(sliceOutput, singleMapBlock.keyType);
+
         int offset = singleMapBlock.getOffset();
         int positionCount = singleMapBlock.getPositionCount();
-        keyBlockEncoding.writeBlock(sliceOutput, singleMapBlock.getKeyBlock().getRegion(offset / 2, positionCount / 2));
-        valueBlockEncoding.writeBlock(sliceOutput, singleMapBlock.getValueBlock().getRegion(offset / 2, positionCount / 2));
+        blockEncodingSerde.writeBlock(sliceOutput, singleMapBlock.getKeyBlock().getRegion(offset / 2, positionCount / 2));
+        blockEncodingSerde.writeBlock(sliceOutput, singleMapBlock.getValueBlock().getRegion(offset / 2, positionCount / 2));
```

Comment:
a helper here would be a big help

Reply:
Done renaming. I will try out introducing the helper separately.